docs/tutorial/t-spark-monitoring.md

```diff
@@ -0,0 +1,350 @@
+
+```bash
+aws s3api put-object --bucket spark-tutorial --key spark-events/ # / at the end is requied
+```
+
+Now, let's deploy the [`spark-history-server-k8s`](https://github.com/canonical/spark-history-server-k8s-operator) charm in our Juju model. 
+
+```bash
+juju deploy spark-history-server-k8s -n1 --channel 3.4/stable
+```
+
+Spark History Server needs to connect to the S3 bucket for it to be able to read the event logs. This integration is provided to Spark History server charm by the [`s3-integrator`](https://github.com/canonical/s3-integrator) charm. Let's deploy `s3-integrator` charm, configure it and integrate it with `spark-history-server-k8s`.
+
+```bash
+# Deploy s3-integrator
+juju deploy s3-integrator -n1 --channel edge
+
+# Configure s3-integrator with correct bucket name, directory path and endpoint
+juju config s3-integrator bucket=spark-tutorial path="spark-events" endpoint=http://$S3_ENDPOINT
+
+# Sync S3 credentials with s3-integrator
+juju run s3-integrator/leader sync-s3-credentials \
+  access-key=$ACCESS_KEY secret-key=$SECRET_KEY
+
+# Integrate s3-integrator with Spark History Server
+juju integrate s3-integrator spark-history-server-k8s
+```
+
+Let's view the status of Juju model at this point with `juju status --watch 1s --relations` command. Once the charms have completed deployment and integration, the status should appear simlar to the following:
+
+
+```
+Model           Controller      Cloud/Region        Version  SLA          Timestamp
+history-server  spark-tutorial  microk8s/localhost  3.1.7    unsupported  07:29:35Z
+
+App                       Version  Status  Scale  Charm                     Channel     Rev  Address         Exposed  Message
+s3-integrator                      active      1  s3-integrator             edge         14  10.152.183.87   no       
+spark-history-server-k8s           active      1  spark-history-server-k8s  3.4/stable   15  10.152.183.111  no       
+
+Unit                         Workload  Agent  Address      Ports  Message
+s3-integrator/0*             active    idle   10.1.29.167         
+spark-history-server-k8s/0*  active    idle   10.1.29.169         
+
+Integration provider               Requirer                                 Interface            Type     Message
+s3-integrator:s3-credentials       spark-history-server-k8s:s3-credentials  s3                   regular  
+s3-integrator:s3-integrator-peers  s3-integrator:s3-integrator-peers        s3-integrator-peers  peer   
+```
+
+The Spark History Server is now successfully configured to read event logs from S3 bucket. Let's run a simple job so that Spark can generate some event logs. We're going to use the same `count_vowels.py` example we used earlier, which already exists in the `spark-tutorial` bucket.
+
+
+```bash
+spark-client.spark-submit \
+    --username spark --namespace history-server \
```

Comment:
It is not necessary to run the job in the same namespace

Reply:
During my testing, using different namespaces resulted in the DNS of the pods (running in Juju model's namespace) not being resolved from the pods running in spark serviceaccount's namespace. That's the reason I kept dedicated namespaces for each section of the tutorial.